src/timedate/timedated.c

```diff
@@ -846,9 +916,11 @@ static int method_set_ntp(sd_bus_message *m, void *userdata, sd_bus_error *error
 
         log_info("Set NTP to %sd", enable_disable(enable));
 
-        (void) sd_bus_emit_properties_changed(bus, "/org/freedesktop/timedate1", "org.freedesktop.timedate1", "NTP", NULL);
+        r = sd_bus_message_new_method_return(m, &c->reply_set_ntp);
+        if (r < 0)
+                return r;
```

Comment:
Hmm, this changes the semantics of `method_set_ntp` significantly — before it'd be "set the state" and now it is "set the state, wait for the unit to be started or stopped". Additional wrinkle is that if a second parallel `method_set_ntp` is made in parallel, the first one will cut the wait short, so the condition is really more like "wait for the unit to be started or stopped OR for another method_set_ntp call to come in". Shouldn't the change be made smaller, to keep the call asynchronous, and only wait on the job to be finished to send out the property change? This should fix the original issue, without changing the semantics of the call. I'm not sure what the proper solution is, so I'm putting this out here for discussion.

Reply:
I confirm that this fixes the wrong property changed value. However, I agree with @keszybz that it seems cleaner to not make `SetNTP` blocking, and just defer the `PropertyChanged` signal until the NTP unit is actually running.

Reply:
Ah, I see. OK, I will update this. Thank you.